src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const initialValue = 0;
+
+  const config = {
+    result: initialValue,
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result /= value;
+    },
+
+    reset() {
+      this.result = initialValue;
+
+      return this;
+    },
+
+    operate(operation, value) {
+      operation.bind(this)(value);
```

Comment:
Consider checking if the `operation` is a function before attempting to bind and call it. This can prevent potential runtime errors if a non-function is passed.